src/commonmark.rs

```diff
@@ -440,6 +440,68 @@ impl<'a, W: Write + 'a> Context<'a, W> {
         );
         write!(self.output.writer, "{}", osc(&command))
     }
+
+    /// Write an inline image for Terminology.
+    fn write_terminology_inline_image(&mut self, path: &Path) -> Result<()> {
+        fn get_lines(columns: u16, file: &str) -> u16 {
+            use std::process::Command;
+            use termion::terminal_size;
+
+            let lines = terminal_size().unwrap_or((0, 24)).1;
+
+            Command::new("identify")
+                .arg("-format")
+                .arg("%w %h")
+                .arg(file)
+                .output()
+                .and_then(|o| {
+                    fn get_v(x: Option<&str>) -> Result<u32> {
+                        x.and_then(|x| x.parse().ok())
+                            .ok_or(Into::<Error>::into(ErrorKind::InvalidData))
+                    }
+
+                    let o = String::from_utf8(o.stdout).or(Err(ErrorKind::InvalidData))?;
+                    let mut tokens = o.split(' ');
+                    let (w, h) = (get_v(tokens.next())? as f64, get_v(tokens.next())? as f64);
```

Comment:
Please `.map` over `tokens` to parse each item and use [`collect`](https://doc.rust-lang.org/1.22.1/std/iter/trait.Iterator.html#method.collect) to extract the result; this avoids the extra helper function.